Core/include/Acts/EventData/Measurement.hpp

```diff
@@ -391,4 +399,232 @@ using FittableMeasurement =
 template <typename source_link_t>
 using FittableVolumeMeasurement =
     typename fittable_volume_measurement_helper<source_link_t>::type;
+
+/// A measurement of a fixed-size subspace of the full parameters.
+///
+/// @tparam source_link_t Source link type to connect to the detector readout
+/// @tparam indices_t Parameter index type, determines the full parameter space
+/// @tparam kSize Size of the parameter subset.
+///
+/// The measurement is templated only on the measurement subspace size; its
+/// content (indices) is defined only at runtime. In previous implementations,
+/// the measurement was templated directly on the subspace indices. This lead to
+/// a combinatorial explosion and severe penalty during compilation since all
+/// possible combinations need to be considered for the measurement variant.
+/// However, for the computations only the size of the measurement subspace is
```

Comment:
I think the history should not be part of the documentation, but of the PR.

Reply:
I added the history so that in the future if somebody comes up and says "Well, why do we not template this on the indices. That looks like a cool way of simplifying this", we can point him to it and not do the same mistake again.